mmrazor/models/losses/kl_divergence.py

```diff
@@ -13,35 +13,49 @@ class KLDivergence(nn.Module):
 
     Args:
         tau (float): Temperature coefficient. Defaults to 1.0.
+        reduction (str): Specifies the reduction to apply to the loss:
+            ``'none'`` | ``'batchmean'`` | ``'sum'`` | ``'mean'``.
+            ``'none'``: no reduction will be applied
+            ``'batchmean'``: the sum of the output will be divided by the batchsize
+            ``'sum'``: the output will be summed
+            ``'mean'``: the output will be divided by the number of elements in the output
+            Default: ``'batchmean'``
         loss_weight (float): Weight of loss. Defaults to 1.0.
     """
 
     def __init__(
         self,
         tau=1.0,
+        reduction='batchmean',
         loss_weight=1.0,
     ):
         super(KLDivergence, self).__init__()
         self.tau = tau
         self.loss_weight = loss_weight
 
+        accept_reduction = {'none', 'batchmean', 'sum', 'mean'}
+        assert reduction in accept_reduction, \
+            f'KLDivergence supports reduction {accept_reduction}, ' \
+            f'but gets {reduction}.'
+        self.reduction = reduction
+
     def forward(self, preds_S, preds_T):
         """Forward computation.
 
         Args:
             preds_S (torch.Tensor): The student model prediction with
-                shape (N, C, W, H).
+                shape (N, num_classes).
```

Comment:
#29 has been merged. There are some conflicts.
Because the `kl_div` may be used in segmentation, the shape could be (N, C, H, W).

Reply:
Fixed.